Alerting docs: `Provision alerting resources` updates (#82221)
grafana/grafana: docs/sources/alerting/set-up/provision-alerting-resources/terraform-provisioning/index.md

  - Terraform
labels:
  products:
    - cloud
    - enterprise
    - oss
menuTitle: Use Terraform to provision
title: Use Terraform to provision alerting resources
weight: 200
---

# Use Terraform to provision alerting resources

Use Terraform’s Grafana Provider to manage your alerting resources and provision them into your Grafana system. Terraform provider support for Grafana Alerting makes it easy to create, manage, and maintain your entire Grafana Alerting stack as code.

> For more examples and information about the Terraform Alerting schemas, refer to the [Grafana Provider](https://registry.terraform.io/providers/grafana/grafana/latest/docs) documentation.

Reviewer: ```suggestion
Refer to [Grafana Provider](https://registry.terraform.io/providers/grafana/grafana/latest/docs) documentation for more examples and information on Terraform Alerting schemas.
```

Complete the following tasks to create and manage your alerting resources using Terraform.

1. Create an API key for provisioning.

> 2. Configure the Terraform provider.
> 3. Define your alerting resources in Terraform. [Export alerting resources][alerting_export] in Terraform format, or implement the [Terraform Alerting schemas](https://registry.terraform.io/providers/grafana/grafana/latest/docs).

Reviewer: ```suggestion
1. Configure the Terraform provider.
1. Define your alerting resources in Terraform. [Export alerting resources][alerting_export] in Terraform format, or implement the [Terraform Alerting schemas](https://registry.terraform.io/providers/grafana/grafana/latest/docs).
```

Reviewer: we use lazy numbering in markdown is always 1. 1. 1. and it renders correctly https://grafana.com/docs/writers-toolkit/write/style-guide/style-conventions/#ordered-lists

   ```HCL
   resource "grafana_notification_policy" "my_policy" {
       group_by = ["alertname"]
       contact_point = grafana_contact_point.my_slack_contact_point.name

       group_wait = "45s"
       group_interval = "6m"
       repeat_interval = "3h"

       policy {
           matcher {
               label = "a"
               match = "="
               value = "b"
           }
           group_by = ["..."]
           contact_point = grafana_contact_point.a_different_contact_point.name
           mute_timings = [grafana_mute_timing.my_mute_timing.name]

           policy {
               matcher {
                   label = "sublabel"
                   match = "="
                   value = "subvalue"
               }
               contact_point = grafana_contact_point.a_third_contact_point.name
               group_by = ["..."]
           }
       }
   }
   ```

2. In the mute_timings field, link a mute timing to your notification policy.

3. Run the command `terraform apply`.

4. Go to the Grafana UI and check the details of your notification policy.

>    **Note:**

Reviewer: Use admonition for note {{< admonition type="note" >}}
{{< /admonition >}}

Reviewer: and lazy numbering for all please:)

Author: fixed by https://github.com/grafana/grafana/pull/82221/commits/abbd39ca219b9e8d137ae46320a4fc0ee7c15456
